ecs-agent/api/resource/ebs_discovery_linux.go

```diff
@@ -36,9 +37,9 @@ type BlockDevice struct {
 
 // ConfirmEBSVolumeIsAttached is used to scan for an EBS volume that's on the host with a specific device name and/or volume ID.
 // There are two cases:
-// 1. On nitro-based instance we check both device name and volume ID.
-// 2. On xen-based instance we only check by the device name.
-func (api *EBSDiscoveryClient) ConfirmEBSVolumeIsAttached(deviceName, volumeID string) error {
+// 1. On nitro-based instance we check by volume ID.
+// 2. On xen-based instance we check by the device name. (TODO)
```

Comment:
can we be sure that we fail this function if we don't have a volumeId in the lsblk or if the deviceName doesn't match up?

Reply:
ah yeah we can add a check if the device name is empty and if it is then we return an error. good call out

Reply:
We're actually already covering this scenario. In the case that we don't find the volume ID on the host, we'll return an empty string as the device name and an error.